VastClient/VastClient/VastTracker.swift

```diff
@@ -9,20 +9,32 @@
 import Foundation
 
 public protocol VastTrackerDelegate {
-    func adBreakStart(_ id: String, _ vastModel: VastModel)
-    func adStart(_ id: String, _ ad: VastAd)
-    func adFirstQuartile(_ id: String, _ ad: VastAd)
-    func adMidpoint(_ id: String, _ ad: VastAd)
-    func adThirdQuartile(_ id: String, _ ad: VastAd)
-    func adComplete(_ id: String, _ ad: VastAd)
-    func adBreakComplete(_ id: String, _ vastModel: VastModel)
+    func adBreakStart(vastTracker: VastTracker, totalAds: Int)
+    func adStart(vastTracker: VastTracker, ad: VastAd)
+    func adFirstQuartile(vastTracker: VastTracker, ad: VastAd)
+    func adMidpoint(vastTracker: VastTracker, ad: VastAd)
+    func adThirdQuartile(vastTracker: VastTracker, ad: VastAd)
+    func adComplete(vastTracker: VastTracker, ad: VastAd)
+    func adBreakComplete(vastTracker: VastTracker, vastModel: VastModel)
+}
+
+enum TrackerModelType {
+    case standAlone
+    case adPod
+    case adBuffet
+}
+
+public struct TrackerModel {
+    let type: TrackerModelType
+    let vastModel: VastModel
 }
 
 public class VastTracker {
 
     public var delegate: VastTrackerDelegate?
+    public let totalAds: Int
 
-    private let id: String
+    let id: String
     private var trackingStatus: TrackingStatus = .unknown
     private let vastModel: VastModel
```

Comment:
could you please make the following properties public on this class:

vastModel
vmapModel
vmapAdBreak